### ISSUE TITLE
Spurious error BS1001?

### ISSUE THREAD

**Issue by ZeeD** [rokucommunity/brighterscript#700]
example snippet:

```brighterscript
sub foo(flag as boolean)
    ' both cases set ret
    if flag
        ret = "foo"
    else
        ret = "bar"
    end if

    print ret
end sub
```

I have updated `brighterscript` from `0.57.2` to `0.58.0` and then now `bsc` triggers the error `error BS1001: Cannot find name 'ret'`. My understanding is that `ret` is actually been set, so `bsc` shouldn't trigger this error

(moreover I found this error in some libraries I use, like https://github.com/rokucommunity/roku-requests/blob/master/src/source/Requests.brs#L177 trigger the same `BS1001` error, so it's a bit complex to fix myself)

**Comment by TwitchBronBron** [comment]
I started https://github.com/rokucommunity/brighterscript/pull/698 to fix this exact issue the other day, but my implementation has a bug. I'm hoping to get it fixed in the next few days. 

For now, you can mitigate the issue in your own code by lifting the `ret` value above the if statement. 

```brighterscript
sub foo(flag as boolean)
    ret = invalid
    ' both cases set ret
    if flag
        ret = "foo"
    else
        ret = "bar"
    end if

    print ret
end sub
```

**Comment by ZeeD** [comment]
I don't know if this is related or not, but I found another different behavior with this new release:

```brighterscript
namespace ns
    sub foo()
        bar()
    end sub

    sub bar()
        ' not important
    end sub
end namespace
```

this snippet now triggers `error BSLINT1001: Using uninitialised variable 'bar' when this file is included in scope 'source'`

(keep in mind that I also use bslint as one of the bsc plugins)
```json5
{
    // ...
    "plugins": [
        "@rokucommunity/bslint"
    ]
}
```

**Comment by TwitchBronBron** [comment]
I'm pretty sure bslint does not have support for namespace relative function calls at this time.

**Comment by elsassph** [comment]
@ZeeD this definitely look like a different issue - could you start a new issue and maybe clarify if there seem to be a regression (something valid which wasn't an error and became an error)

**Comment by TwitchBronBron** [comment]
The original issue (errors when using variables that were defined within if statements) has been resolved in brighterscript [v0.60.2](https://github.com/rokucommunity/brighterscript/blob/master/CHANGELOG.md#0602---2022-10-18)
